src/tests/test_animation/test_animation.cpp

```diff
@@ -383,6 +383,40 @@ private Q_SLOTS:
 		QCOMPARE(a.key_get_frame(2), 40);
 		QCOMPARE(a.key_get_frame(3), -1);
 	}
+
+	void NextKey()
+	{
+		Properties p;
+		p.set("foo", "50=100; 60=60; 100=0");
+		// Cause the string to be interpreted as animated value.
+		p.anim_get_int("foo", 0);
+		Animation a = p.get_animation("foo");
+		QVERIFY(a.is_valid());
+
+		QCOMPARE(a.next_key(0), 50);
+		QCOMPARE(a.next_key(59), 60);
+		QCOMPARE(a.next_key(60), 60);
+		QCOMPARE(a.next_key(61), 100);
+		QCOMPARE(a.next_key(100), 100);
+		QCOMPARE(a.next_key(101), 1);
```

Comment:
This seems like a strange return value for this. It is outside of the range, but maybe it should return an error?

Reply:
Yeah, one thing we need to be careful about when adding a unit test is that we are basically agreeing to the tested behavior.
The docs for `mlt_animation_next_key()` says
>  * \return true if there was an error

However, `Animation::next_key()` does this:

```
	int error = mlt_animation_next_key( instance, &item, position );
	if ( !error )
		return item.frame;
	else
		return error;
```
Thus, there is no way to differentiate currently between 1 as true for error and 1 as the frame number. We could change the return from the C function as -1. Then, it is still true for error and not a valid frame number. -1 is not the MLT convention for true, but this would need to be an exception and documented.

Reply:
> We could change the return from the C function as -1

I thought of this. But then we would have to decide that negative frame values are illegal. Right now it is easy to get negative frame values by setting them negative with mlt_animation_key_set_frame() or mlt_animation_shift_frames().

Another option would be to create a new function that takes a pointer to an integer to be modified and returns a real error code. The current functions could be deprecated.

> However, it might affect Shotcut and be out-of-scope for the next release.

I agree. And maybe the priority is low. Just noticed this inconsistency when debugging and thought I would say something.

If you are inspired by a solution, I am willing to implement, test and integrate with Shotcut. Else, we can leave it as-is for now.

Reply:
Yes, let's create the new methods, but let's make the parameters references instead of pointers to be consistent with the other methods in this class. You can overload these methods with the same name.